profiler/internal/fastdelta/location_index.go

```diff
@@ -0,0 +1,117 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package fastdelta
+
+// locationIndex links location IDs to the addresses, mappings, and function
+// IDs referenced by the location
+type locationIndex struct {
+	// fastpath bookkeeping when locations have 1-up identifiers
+	fastAddress     []uint64
+	fastMappingID   []uint64
+	fastFunctionIDs [][]uint64
+	fastFunctionIdx int
+
+	fallbackAddress     map[uint64]uint64
+	fallbackMappingID   map[uint64]uint64
+	fallbackFunctionIDs map[uint64][]uint64
+}
+
+func (l *locationIndex) reset() {
+	l.fastAddress = l.fastAddress[:0]
+	l.fastMappingID = l.fastMappingID[:0]
+	l.fastFunctionIdx = 0
+
+	l.fallbackAddress = nil
+	l.fallbackMappingID = nil
+	l.fallbackFunctionIDs = nil
+}
+
+// insert associates the given address, mapping ID, and function IDs with the
+// given location ID
+func (l *locationIndex) insert(id, address, mappingID uint64, functionIDs []uint64) {
```

Comment:
it used to live in `fd.go` and I don't remember why it was moved out; @nsrip-dd, do you remember? I don't think we'd ever expose this to outside callers.

Reply:
I moved it to another file because I felt it was a somewhat trickier component that I'd want to look at on its own. As for exposing it to outside callers, Felix's point is that having the methods we actually intend to use capitalized is a reminder for us of what the intended interface is for this type. If this _did_ live in another package, Insert, Get, and GetMeta would be the methods we actually expose. The rest would be implementation details. That's also reflected in the current tests, which only access `locationIndex` through those methods.

Reply:
SGTM; thanks